### DISCUSSION TITLE
Support for Halemeier HA-ZM12/24-4K (4 channel LED switch/dimmer)

### DISCUSSION THREAD

**Discussion by hartmanrik** [Koenkk/zigbee2mqtt#18300] [Devices]
I'm trying to implement the Halemeier HA-ZM12/24-4K Smart receiver12/24VDC 4-channel.
The HA-ZM12/24-1K is already supported (1 channel).

I couldn't find similar devices, so I tried my best to get some things started with the following external converter. 

```javascript
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;


const definition = {
		zigbeeModel: ['HA-ZM12/24-4K'], // The model ID from: Device with modelID 'HA-ZM12/24-4K' is not supported.
		model: 'HA-ZM12/24-4K', // Vendor model number, look on the device for a model number
		vendor: 'Halemeier', // Vendor of the device (only used for documentation and startup logging)
		description: 'Smart receiver  12/24VDC 4-channel', // Description of the device, copy from vendor site. (only used for documentation and startup logging)
		extend: extend.light_onoff_brightness({noConfigure: true}),
		exposes: [e.light().withEndpoint('l1'), e.light().withEndpoint('l2'), e.light().withEndpoint('l3'), e.light().withEndpoint('l4')],
		endpoint: (device) => {
			return {'l1': 1, 'l2': 2, 'l3': 3, 'l4': 4};
		},
		meta: {multiEndpoint: true},
		//whiteLabel: [{vendor: 'Sunricher', model: 'SRP-ZG9105-CC'}],
		configure: async (device, coordinatorEndpoint, logger) => {
			await reporting.bind(device.getEndpoint(1), coordinatorEndpoint, ['genOnOff','genLevelCtrl']);
			await reporting.bind(device.getEndpoint(2), coordinatorEndpoint, ['genOnOff','genLevelCtrl']);
			await reporting.bind(device.getEndpoint(3), coordinatorEndpoint, ['genOnOff','genLevelCtrl']);
			await reporting.bind(device.getEndpoint(4), coordinatorEndpoint, ['genOnOff','genLevelCtrl']);
		
		}};

module.exports = definition;
```
The idea is to get 4 different lights with on off brightness controls  (4 different endpoints)
Unfortunately it is not working as intended, the main problem is my lack of knowledge on how to implement multiple endpoints.
I'm not able to control the brightness and the light is displayed as a switch.

Could someone point me to the right direction?

Thanks in advance!

Exposes
![zigbee-exposes](https://github.com/Koenkk/zigbee2mqtt/assets/73544149/085b9cea-dffd-40a4-9fd8-79c23386f2b0)

Cluster
![zigbee-cluster](https://github.com/Koenkk/zigbee2mqtt/assets/73544149/cc96ad9f-503a-496e-9128-653298786484)

States
![zigbee-state](https://github.com/Koenkk/zigbee2mqtt/assets/73544149/c8378cf2-c77e-4e19-9255-e53623bc052a)

**Comment by Koenkk** [comment]
Instead of `e.light().withEndpoint('l1')` use `e.light_brightness().withEndpoint('l1')`

**Comment by hartmanrik** [comment]
> ```js
> Smart receiver  12/24VDC 4-channel
> ```

How easy can it be, it's working now, thanks so much!

**Comment by Koenkk** [comment]
Once everything works, please submit a pull request (then this device can be supported out-of-the-box)

**Comment by Waakvlam** [comment]
I have this 4-channel receiver. This device is not supported yet out-of-the-box. What can I do about that?

**Reply by andrei-lazarov** [reply]
Please open an issue and fill the template.
Provide the generated external definition (device -> dev console)
